### DISCUSSION TITLE
Grasping Issue

### DISCUSSION THREAD

**Discussion by Muhammad441** [bulletphysics/bullet3#3398] [General]
Hi, 

I am trying to grasp small blocks in pybullet using the PR2 gripper. However, I notice that there are some artifacts that I encounter while actually grasping the objects. This leads to significant inconsistencies. I have attached a gif of one such artifact. Can someone confirm the source of this issue and possible solutions?

In the example below, the gripper is closed by position controlling to a pre-determined value. There are also cases, where the gripper does not properly grasp the object causing it to fall. 

PR2 model source: https://github.com/caelan/pybullet-planning.git

![ezgif com-gif-maker](https://user-images.githubusercontent.com/11947519/104827734-155c2680-582f-11eb-9be4-1a2b6ae35a43.gif)

**Comment by erwincoumans** [comment]
If there would be a separate 50 lines reproduction in a simple PyBullet script, I may look at it, but can't start digging into various reposities.
